uxarray/core/dataset.py

```diff
@@ -327,6 +327,61 @@ def from_healpix(
 
         return cls.from_xarray(ds, uxgrid, {face_dim: "n_face"})
 
+    @classmethod
+    def _wrap_groupby_result(
+        cls, result, original_uxgrid, original_source_datasets=None
+    ):
+        """Factory method to wrap groupby results with uxgrid preservation."""
+        if isinstance(result, xr.Dataset):
+            return cls(
+                result, uxgrid=original_uxgrid, source_datasets=original_source_datasets
+            )
+        elif isinstance(result, xr.DataArray):
+            from uxarray.core.dataarray import UxDataArray
+
+            return UxDataArray(result, uxgrid=original_uxgrid)
+        return result
+
+    def groupby(self, group, squeeze: bool = False, restore_coord_dims: bool = None):
+        """Override to return a groupby object that preserves uxgrid on aggregation."""
```

Comment:
Instead of creating a new docstring, let's use the original docstring from the Xarray method. See an example of how we did it here:

https://github.com/UXARRAY/uxarray/blob/61ff40a67b9d36e1a2c045afbfee275f84819150/uxarray/core/dataset.py#L522-L530